In [1]:
#libreria utiles 
import pandas as pd
import numpy as np
from scipy import stats
from matplotlib import pyplot as plt

## Organizar el DataFrame para hacer los analisis de varianza 

In [2]:
# Cargar el archivo Excel
SCA = pd.read_excel("/home/camilo/Documentos/Doctorado en Matemática/TrabajoCruzCarreto/ArticuloMultinomialSCA2/Base de datos COMPLETA 230822.xls", sheet_name = "GRUPO ENFERMOS")
Control = pd.read_excel("/home/camilo/Documentos/Doctorado en Matemática/TrabajoCruzCarreto/ArticuloMultinomialSCA2/Base de datos COMPLETA 230822.xls", sheet_name = "Grupo de Control")
Preclinics = pd.read_excel("/home/camilo/Documentos/Doctorado en Matemática/TrabajoCruzCarreto/ArticuloMultinomialSCA2/Base de datos COMPLETA 230822.xls", sheet_name = "GRUPO PRECLINICOS")

In [3]:
#Preprosesamiento
# Crear una nueva columna 'padecimiento' con el valor "E" repetido para todas las filas
SCA['padecimiento'] = np.repeat("E", SCA.shape[0])
Control['padecimiento'] = np.repeat("C", Control.shape[0])
Preclinics['padecimiento'] = np.repeat("P", Preclinics.shape[0])

In [4]:
# Filtrar el DataFrame SCA donde las columnas 'DAPMES', 'Columna2', 'Columna3' no tengan valores nulos
SCA_l = SCA.dropna(subset=['DAPMES', 'DAPP', 'DAPM', 'DAPCM', 'DAPIV', 'DAP' ])

# Filtrar el DataFrame Control donde las columnas 'DAPMES', 'Columna2', 'Columna3' no tengan valores nulos
Control_l = Control.dropna(subset=['DAPMES', 'DAPP', 'DAPM', 'DAPCM', 'DAPIV', 'DAP' ])

# Filtrar el DataFrame Preclinicos donde las columnas 'DAPMES', 'Columna2', 'Columna3' no tengan valores nulos
Preclinics_l = Preclinics.dropna(subset=['DAPMES', 'DAPP', 'DAPM', 'DAPCM', 'DAPIV', 'DAP' ])

In [5]:
# Crear una lista de IDs en tu DataFrame SCA_l
id_SCA_l = np.arange(len(SCA_l))

# Seleccionar 13 IDs aleatorios sin reemplazo
ID_test = np.random.choice(id_SCA_l, 13, replace=False)

# Crear el conjunto de prueba utilizando los IDs seleccionados
SCA_l_test = SCA_l.iloc[ID_test]

# Crear el conjunto de entrenamiento excluyendo los índices seleccionados
SCA_l_train = SCA_l.drop(SCA_l.index[ID_test])

# Crear una lista de IDs en tu DataFrame Control_l
id_Control_l = np.arange(len(Control_l))

# Seleccionar 14 IDs aleatorios sin reemplazo
ID_test = np.random.choice(id_Control_l, 14, replace=False)

# Crear el conjunto de prueba utilizando los IDs seleccionados
Control_l_test = Control_l.iloc[ID_test]

# Crear el conjunto de entrenamiento excluyendo los índices seleccionados
Control_l_train = Control_l.drop(Control_l.index[ID_test])

# Crear una lista de IDs en tu DataFrame Control_l
id_Preclinics_l = np.arange(len(Preclinics_l))

# Seleccionar 14 IDs aleatorios sin reemplazo
ID_test = np.random.choice(id_Preclinics_l, 4, replace=False)

# Crear el conjunto de prueba utilizando los IDs seleccionados
Preclinics_l_test = Preclinics_l.iloc[ID_test]

# Crear el conjunto de entrenamiento excluyendo los índices seleccionados
Preclinics_l_train = Preclinics_l.drop(Preclinics_l.index[ID_test])

In [48]:
# Y los DataFrames correspondientes para test: SCA_l_test, Preclinicos_l_test, Control_l_test

# Merge para el conjunto de entrenamiento
dataset_train = pd.merge(SCA_l_train, Preclinics_l_train, how='outer')
dataset_train = pd.merge(dataset_train, Control_l_train, how='outer')

# Mostrar el conjunto de entrenamiento combinado
print("Conjunto de entrenamiento combinado:")
print(dataset_train)

# Merge para el conjunto de prueba
dataset_test = pd.merge(SCA_l_test, Preclinics_l_test, how='outer')
dataset_test = pd.merge(dataset_test, Control_l_test, how='outer')

dataset_full = pd.merge(dataset_train, dataset_test, how='outer')

Conjunto de entrenamiento combinado:
              No  Nombres y Apellidos / IRM No? Sexo  Edad Edad.1 Tiempo  \
0    IIRM22-1693                            NaN    M  35.0    NaN    NaN   
1     IRM21-1598                            NaN    F  32.0    NaN    NaN   
2     IRM21-1777             GOMEZ FONT^DANIEL^    M  45.0     41      4   
3     IRM21-1840  GONZALEZ HECHAVARRIA^IDALMIS^    F  58.0     34     24   
4     IRM21-1859         MARTIN INFANTE^MIRTHA^    F  70.0     60     10   
..           ...                            ...  ...   ...    ...    ...   
118    IRM22-872                            NaN    F  50.0    NaN    NaN   
119    IRM22-890     PEREZ HECHEVARRIA^NERIO^^^    M  70.0     45     25   
120    IRM22-944      OSORIO RAMIREZ^MAGDIEL^^^    M  25.0     17      8   
121    IRM22-946       RIVAS MEDINA^LOURDES ^^^    F  47.0     43      4   
122     IRM22-95           NORAIDA SOLIS GARCIA    F  43.0     42      1   

                 SARA                             

In [7]:
print(dataset_full)

              No  Nombres y Apellidos / IRM No? Sexo  Edad Edad.1 Tiempo  \
0    IIRM22-1693                            NaN    M  35.0    NaN    NaN   
1     IRM21-1598                            NaN    F  32.0    NaN    NaN   
2     IRM21-1777             GOMEZ FONT^DANIEL^    M  45.0     41      4   
3     IRM21-1840  GONZALEZ HECHAVARRIA^IDALMIS^    F  58.0     34     24   
4     IRM21-1859         MARTIN INFANTE^MIRTHA^    F  70.0     60     10   
..           ...                            ...  ...   ...    ...    ...   
149    IRM22-944      OSORIO RAMIREZ^MAGDIEL^^^    M  25.0     17      8   
150    IRM22-945       BRUSETAS RIVAS^GLADIS^^^    F  62.0     47     15   
151    IRM22-946       RIVAS MEDINA^LOURDES ^^^    F  47.0     43      4   
152     IRM22-95           NORAIDA SOLIS GARCIA    F  43.0     42      1   
153    IRM22-980         MORENO TORRES^PEDRO^^^    M  64.0     58      8   

                 SARA                                Fenotipo  DAPMES   DAPP  \
0      

In [9]:
dataset_full=dataset_full[['Edad','Sexo','DAPMES','DAPP','DAPM','DAPCM','padecimiento']]
print(dataset_full)

     Edad Sexo  DAPMES   DAPP  DAPM  DAPCM padecimiento
0    35.0    M   19.10  23.93  6.72  11.81            C
1    32.0    F   19.17  24.36  7.70  12.55            C
2    45.0    M   15.74  15.38  6.09  12.64            E
3    58.0    F   17.97  16.24  5.80  11.59            E
4    70.0    F   17.20  21.56  6.28  12.12            E
..    ...  ...     ...    ...   ...    ...          ...
149  25.0    M   16.22  15.71  5.31  11.14            E
150  62.0    F   15.42  18.44  5.32  13.32            E
151  47.0    F   15.92  17.80  4.85   8.70            E
152  43.0    F   15.45  20.25  5.99  10.86            E
153  64.0    M   15.57  14.51  6.35   9.84            E

[154 rows x 7 columns]


In [10]:
dataset_full.isnull().sum()

Edad            2
Sexo            0
DAPMES          0
DAPP            0
DAPM            0
DAPCM           0
padecimiento    0
dtype: int64

In [11]:
dataset_full['padecimiento'].value_counts()

padecimiento
C    72
E    65
P    17
Name: count, dtype: int64

## Prueba para conocer si hay diferencias Significativas entre Control, Enfermo(SCA), Preclinico acerca del Diametro anteposterior del Mesencefalo DAPMES

## Opción 1: ANOVA de una vía (One-Way ANOVA) con scipy
Esta opción se usa si tienes una variable dependiente continua y una categórica con varios niveles (grupos).

In [17]:
C=dataset_full[dataset_full['padecimiento']=='C']['DAPMES'].values
E=dataset_full[dataset_full['padecimiento']=='E']['DAPMES'].values
P=dataset_full[dataset_full['padecimiento']=='P']['DAPMES'].values

In [18]:
# Realizar el ANOVA
f_stat, p_value = stats.f_oneway(C,E,P)

print(f'Estadístico F: {f_stat:.4f}')
print(f'Valor p: {p_value:.4f}')

Estadístico F: 122.7078
Valor p: 0.0000


###  Estadístico F (F-Statistic)

El estadístico F es una medida que compara la variabilidad entre las medias de los grupos con la variabilidad dentro de los grupos. Su fórmula general es:

F= Variabilidad entre grupos/ Variabilidad dentro de los grupos

Valores altos de F sugieren que las medias de los grupos son muy diferentes entre sí, lo que indica una posible diferencia significativa.

Valores bajos de F sugieren que las medias de los grupos no varían mucho más que la variabilidad aleatoria dentro de los grupos.
En tu caso:

### Valor p (p-value)

El valor p indica la probabilidad de obtener un estadístico F tan extremo (o más) si la hipótesis nula es cierta. La hipótesis nula en ANOVA es que no hay diferencias significativas entre las medias de los grupos.

Si p ≤ 0.05: Rechazamos la hipótesis nula, lo que sugiere que al menos un grupo es significativamente diferente.
Si p > 0.05: No hay evidencia suficiente para rechazar la hipótesis nula, lo que sugiere que las diferencias entre las medias son probablemente debidas al azar.



## Resumen del DAPMES

### Estadístico F (F-Statistic)
F = 122.7078

Un valor alto del estadístico F indica que la variabilidad entre las medias de los grupos es mucho mayor que la variabilidad interna dentro de los grupos.
Esto sugiere que las medias de los grupos son muy diferentes entre sí.

### Valor p (p-value)

p = 0.0000 (o muy cercano a 0)

El valor p indica la probabilidad de obtener un estadístico F tan grande por azar si la hipótesis nula (H₀) es verdadera (es decir, si las medias de los grupos son iguales).
Como p < 0.05, rechazamos la hipótesis nula. Esto sugiere que al menos un grupo es significativamente diferente de los demás.

## Siguiente Paso

Los resultados muestran que hay diferencias significativas entre las medias de los grupos. Ahora sabes que las medias no son todas iguales, pero el ANOVA no indica entre qué grupos hay diferencias.

Siguiente paso: Prueba post hoc (Tukey HSD)
Para identificar qué pares de grupos son diferentes, debes realizar una prueba post hoc como Tukey HSD. 

## Usando la libreria stastmodel

In [20]:
import statsmodels.api as sm
from statsmodels.formula.api import ols

# Definir el modelo (ANOVA de una vía)
modelo = ols('DAPMES ~ padecimiento', data=dataset_full).fit()

# Realizar el ANOVA
anova_tabla = sm.stats.anova_lm(modelo, typ=2)

print(anova_tabla)

                  sum_sq     df           F        PR(>F)
padecimiento  367.718320    2.0  122.707845  2.250979e-32
Residual      226.250677  151.0         NaN           NaN


## Implementación de Tukey HSD en Python
A continuación, te muestro cómo hacerlo utilizando statsmodels.

Ejemplo: Prueba Tukey HSD

In [21]:
import pandas as pd
from statsmodels.stats.multicomp import pairwise_tukeyhsd

# Realizar la prueba Tukey HSD
tukey = pairwise_tukeyhsd(endog=dataset_full['DAPMES'], groups=dataset_full['padecimiento'], alpha=0.05)

# Mostrar los resultados
print(tukey)


Multiple Comparison of Means - Tukey HSD, FWER=0.05 
group1 group2 meandiff p-adj   lower   upper  reject
----------------------------------------------------
     C      E  -3.2726    0.0 -3.7683 -2.7769   True
     C      P  -1.9035    0.0 -2.6848 -1.1222   True
     E      P   1.3691 0.0002  0.5798  2.1584   True
----------------------------------------------------


   Multiple Comparison of Means - Tukey HSD, FWER=0.05
======================================================
group1 group2 meandiff p-adj  lower  upper  reject
------------------------------------------------------
     A      B     5.00 0.743  -21.4  31.4  False
     A      C   -20.00 0.032 -38.37 -1.63   True
     B      C   -25.00 0.013 -44.04 -5.96   True
------------------------------------------------------

## Interpretación de la salida:
meandiff: Diferencia entre las medias de los dos grupos comparados.
p-adj: Valor p ajustado para cada comparación (usando Tukey HSD).
reject: True si hay una diferencia significativa entre los dos grupos, False si no.
En el ejemplo:

No hay diferencias significativas entre los grupos A y B (reject = False).
Sí hay diferencias significativas entre los grupos A y C (reject = True).
También hay diferencias significativas entre B y C (reject = True).


## Prueba para conocer si hay diferencias Significativas entre Control, Enfermo(SCA), Preclinico acerca del Diámetro anteroposterior de protuberancia DAPP


## Anova de una Vía DAPP

In [27]:
C=dataset_full[dataset_full['padecimiento']=='C']['DAPP'].values
E=dataset_full[dataset_full['padecimiento']=='E']['DAPP'].values
P=dataset_full[dataset_full['padecimiento']=='P']['DAPP'].values

In [28]:
# Realizar el ANOVA
f_stat, p_value = stats.f_oneway(C,E,P)

print(f'Estadístico F: {f_stat:.4f}')
print(f'Valor p: {p_value:.4f}')

Estadístico F: 0.7939
Valor p: 0.4540


Resumen:

Estadístico F: 0.7939 → Baja relación entre la variabilidad entre y dentro de los grupos.

Valor p: 0.4540 → No hay diferencias significativas entre las medias de los grupos.

## ANOVA y TUKEY HSD con Stastmodel

In [29]:
# Definir el modelo (ANOVA de una vía)
modelo = ols('DAPP ~ padecimiento', data=dataset_full).fit()

# Realizar el ANOVA
anova_tabla = sm.stats.anova_lm(modelo, typ=2)

print(anova_tabla)

                    sum_sq     df         F   PR(>F)
padecimiento  5.369438e+04    2.0  0.793891  0.45396
Residual      5.106402e+06  151.0       NaN      NaN


In [31]:
# Realizar la prueba Tukey HSD
tukey = pairwise_tukeyhsd(endog=dataset_full['DAPP'], groups=dataset_full['padecimiento'], alpha=0.05)

# Mostrar los resultados
print(tukey)


  Multiple Comparison of Means - Tukey HSD, FWER=0.05  
group1 group2 meandiff p-adj    lower    upper   reject
-------------------------------------------------------
     C      E -38.0958 0.4486 -112.5714  36.3799  False
     C      P -34.5749 0.7655 -151.9514  82.8016  False
     E      P   3.5209 0.9973 -115.0567 122.0985  False
-------------------------------------------------------


### Resumen 

Lamentablemente DAPP no nos va servir segun el ANOVA una VIA y lo confirma el Tukey HSD

## Prueba para conocer si hay diferencias Significativas entre Control, Enfermo(SCA), Preclinico acerca del Diámetro anteroposterior de médula cervical a nivel de C2 DAPM

### Anova

In [42]:
C=dataset_full[dataset_full['padecimiento']=='C']['DAPM'].values
E=dataset_full[dataset_full['padecimiento']=='E']['DAPM'].values
P=dataset_full[dataset_full['padecimiento']=='P']['DAPM'].values

In [43]:
# Realizar el ANOVA
f_stat, p_value = stats.f_oneway(C,E,P)

print(f'Estadístico F: {f_stat:.4f}')
print(f'Valor p: {p_value:.4f}')

Estadístico F: 67.0891
Valor p: 0.0000


## ANOVA y TUKEY HSD con Stastmodel DAPM

In [38]:
# Definir el modelo (ANOVA de una vía)
modelo = ols('DAPM ~ padecimiento', data=dataset_full).fit()

# Realizar el ANOVA
anova_tabla = sm.stats.anova_lm(modelo, typ=2)

print(anova_tabla)

                  sum_sq     df          F        PR(>F)
padecimiento  112.022405    2.0  67.089086  1.417360e-21
Residual      126.066580  151.0        NaN           NaN


In [39]:
# Realizar la prueba Tukey HSD
tukey = pairwise_tukeyhsd(endog=dataset_full['DAPM'], groups=dataset_full['padecimiento'], alpha=0.05)

# Mostrar los resultados
print(tukey)

Multiple Comparison of Means - Tukey HSD, FWER=0.05 
group1 group2 meandiff p-adj   lower   upper  reject
----------------------------------------------------
     C      E  -1.8109    0.0 -2.1809 -1.4408   True
     C      P  -0.8579 0.0019 -1.4411 -0.2747   True
     E      P    0.953 0.0005  0.3638  1.5422   True
----------------------------------------------------


Exelente por parte del DAPM 

## Prueba para conocer si hay diferencias Significativas entre Control, Enfermo(SCA), Preclinico acerca del  DAPCM

In [ ]:
C=dataset_full[dataset_full['padecimiento']=='C']['DAPCM'].values
E=dataset_full[dataset_full['padecimiento']=='E']['DAPCM'].values
P=dataset_full[dataset_full['padecimiento']=='P']['DAPCM'].values

In [ ]:
# Realizar el ANOVA
f_stat, p_value = stats.f_oneway(C,E,P)

print(f'Estadístico F: {f_stat:.4f}')
print(f'Valor p: {p_value:.4f}')

### Estadístico F (F-Statistic)
F = 0.2803 

Un valor bajo del estadístico F indica que la variabilidad entre las medias de los grupos es similar (o menor) que la variabilidad interna dentro de los grupos.
Esto sugiere que las diferencias entre las medias de los grupos son pequeñas o inexistentes.

### Valor p (p-value)

p = 0.7560:

El valor p representa la probabilidad de observar un estadístico F tan bajo (o más extremo) si la hipótesis nula (H₀) es cierta.


En este caso, la hipótesis nula establece que las medias de los grupos son iguales.
p > 0.05



Como p = 0.7560 es mucho mayor que 0.05, no rechazamos la hipótesis nula.
No hay evidencia estadística suficiente para afirmar que existen diferencias significativas entre las medias de los grupos.

In [45]:
#STASTMODEL
# Definir el modelo (ANOVA de una vía)
modelo = ols('DAPCM ~ padecimiento', data=dataset_full).fit()

# Realizar el ANOVA
anova_tabla = sm.stats.anova_lm(modelo, typ=2)

print(anova_tabla)

                  sum_sq     df         F    PR(>F)
padecimiento    1.022315    2.0  0.280274  0.755969
Residual      275.390862  151.0       NaN       NaN


In [46]:
# Realizar la prueba Tukey HSD
tukey = pairwise_tukeyhsd(endog=dataset_full['DAPCM'], groups=dataset_full['padecimiento'], alpha=0.05)

# Mostrar los resultados
print(tukey)

Multiple Comparison of Means - Tukey HSD, FWER=0.05
group1 group2 meandiff p-adj   lower  upper  reject
---------------------------------------------------
     C      E  -0.0785 0.9383 -0.6255 0.4684  False
     C      P   0.1944 0.8548 -0.6676 1.0564  False
     E      P   0.2729  0.739 -0.5979 1.1437  False
---------------------------------------------------


## DAPCM tampoco es útil 

In [49]:
dataset_full.columns

Index(['No', 'Nombres y Apellidos / IRM No?', 'Sexo', 'Edad', 'Edad.1',
       'Tiempo', 'SARA', 'Fenotipo', 'DAPMES', 'DAPP', 'DAPM', 'DAPCM',
       'CoDM (DAPCM / DAPM )', 'DAPIV', 'DHIV', 'DTIV', 'VIV', 'DAP', 'H',
       'CAG Rango Alelo 1', 'CAG Rango Alelo 2', 'Unnamed: 21', 'padecimiento',
       'Nombre y apellidos', 'CAG alelo 1', 'CAG alelo 2'],
      dtype='object')

## Pruebas más directas para las otras variables 

## DAPIV

In [50]:
#STASTMODEL
# Definir el modelo (ANOVA de una vía)
modelo = ols('DAPIV ~ padecimiento', data=dataset_full).fit()

# Realizar el ANOVA
anova_tabla = sm.stats.anova_lm(modelo, typ=2)

print(anova_tabla)

                   sum_sq     df          F    PR(>F)
padecimiento   177.956959    2.0  11.442473  0.000024
Residual      1174.199839  151.0        NaN       NaN


In [51]:
# Realizar la prueba Tukey HSD
tukey = pairwise_tukeyhsd(endog=dataset_full['DAPIV'], groups=dataset_full['padecimiento'], alpha=0.05)

# Mostrar los resultados
print(tukey)

Multiple Comparison of Means - Tukey HSD, FWER=0.05
group1 group2 meandiff p-adj   lower  upper  reject
---------------------------------------------------
     C      E   2.1479    0.0  1.0185 3.2772   True
     C      P  -0.1412 0.9807 -1.9211 1.6387  False
     E      P  -2.2891 0.0085 -4.0872 -0.491   True
---------------------------------------------------


## DHIV

In [53]:
#STASTMODEL
# Definir el modelo (ANOVA de una vía)
modelo = ols('DHIV ~ padecimiento', data=dataset_full).fit()

# Realizar el ANOVA
anova_tabla = sm.stats.anova_lm(modelo, typ=2)

print(anova_tabla)

                   sum_sq     df         F   PR(>F)
padecimiento    91.613815    2.0  3.942431  0.02143
Residual      1754.461291  151.0       NaN      NaN


In [54]:
# Realizar la prueba Tukey HSD
tukey = pairwise_tukeyhsd(endog=dataset_full['DHIV'], groups=dataset_full['padecimiento'], alpha=0.05)

# Mostrar los resultados
print(tukey)

Multiple Comparison of Means - Tukey HSD, FWER=0.05
group1 group2 meandiff p-adj   lower  upper  reject
---------------------------------------------------
     C      E   1.3287  0.062 -0.0518 2.7091  False
     C      P  -0.8084  0.654 -2.9841 1.3673  False
     E      P  -2.1371 0.0586  -4.335 0.0609  False
---------------------------------------------------


### no sirve el DHIV

## DTIV

In [55]:
#STASTMODEL
# Definir el modelo (ANOVA de una vía)
modelo = ols('DTIV ~ padecimiento', data=dataset_full).fit()

# Realizar el ANOVA
anova_tabla = sm.stats.anova_lm(modelo, typ=2)

print(anova_tabla)

                   sum_sq     df          F        PR(>F)
padecimiento  1440.790269    2.0  64.556052  5.485331e-21
Residual      1685.042089  151.0        NaN           NaN


In [56]:
# Realizar la prueba Tukey HSD
tukey = pairwise_tukeyhsd(endog=dataset_full['DHIV'], groups=dataset_full['padecimiento'], alpha=0.05)

# Mostrar los resultados
print(tukey)

Multiple Comparison of Means - Tukey HSD, FWER=0.05
group1 group2 meandiff p-adj   lower  upper  reject
---------------------------------------------------
     C      E   1.3287  0.062 -0.0518 2.7091  False
     C      P  -0.8084  0.654 -2.9841 1.3673  False
     E      P  -2.1371 0.0586  -4.335 0.0609  False
---------------------------------------------------


## DAP y VIV no funcionaron por problema de el tipo de datos 

## ANOVA de DOS VIAS SEXO PADECIMIENTO

In [59]:
import pandas as pd
import statsmodels.api as sm
from statsmodels.formula.api import ols


In [61]:
# Definir el modelo de ANOVA de dos vías
modelo = ols('DAPMES ~ Sexo + padecimiento + Sexo:padecimiento', data=dataset_full).fit()

# Realizar ANOVA de dos vías
anova_tabla = sm.stats.anova_lm(modelo, typ=2)

# Mostrar los resultados
print(anova_tabla)

                       sum_sq     df           F        PR(>F)
Sexo                 2.446011    1.0    1.624640  2.044436e-01
padecimiento       366.161107    2.0  121.602041  5.774534e-32
Sexo:padecimiento    0.980102    2.0    0.325492  7.226877e-01
Residual           222.824564  148.0         NaN           NaN


## Interpretación




In [64]:
#                          sum_sq    df         F    PR(>F)
# C(Sexo)                 105.3333   1.0   0.3941   0.5410
# C(Padecimiento)         784.6667   2.0   1.4713   0.2765
# C(Sexo):C(Padecimiento) 245.3333   2.0   0.4594   0.6445
# Residual               3201.0000  10.0      NaN      NaN


Interpretación de los Resultados:

C(Sexo):


El valor p = 0.5410 indica que el sexo no tiene un efecto significativo sobre la puntuación.
C(Padecimiento):


El valor p = 0.2765 sugiere que el tipo de padecimiento tampoco tiene un efecto significativo sobre la puntuación.
Interacción (Sexo
):


El valor p = 0.6445 indica que no hay interacción significativa entre el sexo y el padecimiento. Es decir, el efecto de uno no depende del otro.


Conclusión:
No hay diferencias significativas en las puntuaciones entre los sexos o los padecimientos.
Tampoco se observa una interacción significativa entre el sexo y el padecimiento.
Prueba post hoc si es necesaria:
Si alguno de los efectos hubiera sido significativo (p < 0.05), podrías aplicar una prueba Tukey HSD para identificar específicamente qué categorías difieren.

## Para este resultado se tiene que 

Resultados:

Efecto del Sexo:
Estadístico F = 1.6246
Valor p = 0.2044

Como el valor p > 0.05, no hay un efecto significativo del sexo sobre la variable dependiente (puntuación).
Esto sugiere que las puntuaciones medias no difieren de manera significativa entre hombres (M) y mujeres (F).
Efecto del Padecimiento:

Estadístico F = 121.6020
Valor p = 5.77e-32 (≈ 0.0000)

El valor p < 0.05 indica que hay una diferencia significativa entre los diferentes tipos de padecimiento (C, E y P).
Esto sugiere que las puntuaciones dependen significativamente del tipo de padecimiento.
Interacción entre Sexo y Padecimiento:

Estadístico F = 0.3255
Valor p = 0.7227

Como el valor p > 0.05, no hay una interacción significativa entre el sexo y el padecimiento.
Esto significa que el efecto del sexo no cambia en función del tipo de padecimiento. Por ejemplo, ser hombre o mujer no modifica cómo los diferentes padecimientos afectan las puntuaciones.

Residual:
El residual representa la variabilidad no explicada por los efectos principales ni la interacción.
Esta variabilidad es el "ruido" o aleatoriedad que queda después de considerar los factores analizados.

Conclusión:
El sexo no tiene un efecto significativo sobre las puntuaciones.
El tipo de padecimiento tiene un efecto significativo (p < 0.05), por lo que al menos un padecimiento tiene una media diferente.
No hay interacción significativa entre el sexo y el tipo de padecimiento, lo que significa que ambos factores actúan de manera independiente.

# Próximo paso:
Dado que el tipo de padecimiento tiene un efecto significativo, puedes realizar una prueba post hoc (Tukey HSD) para ver qué pares de padecimientos tienen diferencias significativas. 

In [67]:
from statsmodels.stats.multicomp import pairwise_tukeyhsd

# Realizar la prueba Tukey HSD para los padecimientos
tukey = pairwise_tukeyhsd(endog=dataset_full['DAPMES'], groups=dataset_full['padecimiento'], alpha=0.05)

# Mostrar los resultados
print(tukey)

Multiple Comparison of Means - Tukey HSD, FWER=0.05 
group1 group2 meandiff p-adj   lower   upper  reject
----------------------------------------------------
     C      E  -3.2726    0.0 -3.7683 -2.7769   True
     C      P  -1.9035    0.0 -2.6848 -1.1222   True
     E      P   1.3691 0.0002  0.5798  2.1584   True
----------------------------------------------------


# Interpretación de la Tabla:


group1 y group2:


Estos son los dos grupos comparados en cada fila. En tu caso, los grupos son los tipos de padecimiento: C, E y P.
meandiff:


Diferencia entre las medias de los dos grupos comparados.
Por ejemplo, entre C y E, la diferencia es -3.2726, lo que indica que las puntuaciones del grupo C son menores que las del grupo E.


p-adj:

Valor p ajustado usando el método de Tukey para controlar el error de familia (FWER).
Si p-adj < 0.05, la diferencia entre los dos grupos es significativa. En tu caso, todos los valores p son muy bajos (0.0 o 0.0002), indicando diferencias significativas entre los grupos.
lower y upper:



Intervalo de confianza del 95% para la diferencia entre las medias.
Por ejemplo, para la comparación entre C y E, el intervalo es [-3.7683, -2.7769]. Como este intervalo no incluye el 0, confirma que hay una diferencia significativa.
reject:

True significa que la diferencia entre las medias es estadísticamente significativa (p < 0.05).
False (no se da en tu tabla) indicaría que no hay diferencia significativa entre los dos grupos.
Resultados Resumidos:
C vs E:



Diferencia significativa: El grupo C tiene una media más baja que el grupo E (p < 0.05).
C vs P:



Diferencia significativa: El grupo C tiene una media más baja que el grupo P (p < 0.05).
E vs P:



Diferencia significativa: El grupo E tiene una media mayor que el grupo P (p < 0.05).


## Conclusión:


Existen diferencias significativas entre todas las combinaciones de grupos:
C es diferente tanto de E como de P.
E también es diferente de P.
Esto sugiere que los tres tipos de padecimiento tienen puntuaciones medias significativamente diferentes entre sí.

# LO MISMO SOBRE DAPM

In [69]:
from statsmodels.stats.multicomp import pairwise_tukeyhsd

# Realizar la prueba Tukey HSD para los padecimientos
tukey = pairwise_tukeyhsd(endog=dataset_full['DAPM'], groups=dataset_full['padecimiento'], alpha=0.05)

# Mostrar los resultados
print(tukey)

Multiple Comparison of Means - Tukey HSD, FWER=0.05 
group1 group2 meandiff p-adj   lower   upper  reject
----------------------------------------------------
     C      E  -1.8109    0.0 -2.1809 -1.4408   True
     C      P  -0.8579 0.0019 -1.4411 -0.2747   True
     E      P    0.953 0.0005  0.3638  1.5422   True
----------------------------------------------------


# Interpretación:

group1 y group2:

Los dos grupos comparados en cada fila: C, E y P (padecimientos).
meandiff:

Diferencia entre las medias de los dos grupos comparados.
C vs E: -1.8109 → El grupo C tiene una media 1.81 puntos menor que el grupo E.
C vs P: -0.8579 → El grupo C tiene una media 0.86 puntos menor que el grupo P.
E vs P: 0.953 → El grupo E tiene una media 0.95 puntos mayor que el grupo P.

### p-adj:

Valor p ajustado para corregir el error de familia (FWER).
Todos los valores son menores a 0.05, lo que indica que todas las diferencias son estadísticamente significativas.
lower y upper:

Intervalo de confianza del 95% para la diferencia entre medias. Si el intervalo no incluye el 0, la diferencia es significativa.
Por ejemplo, para C vs E, el intervalo es [-2.1809, -1.4408], lo que confirma una diferencia significativa.


### reject:

True indica que la diferencia entre las medias es estadísticamente significativa (p < 0.05).
Resumen de Resultados:
C vs E:

La media del grupo C es significativamente menor que la del grupo E.
C vs P:

La media del grupo C es significativamente menor que la del grupo P.
E vs P:

La media del grupo E es significativamente mayor que la del grupo P.


### Conclusión:
Todas las comparaciones son significativas: Las medias de los tres grupos (C, E, P) son diferentes entre sí.
Estos resultados sugieren que cada tipo de padecimiento afecta las puntuaciones de manera diferente.